core/templates/services/playthrough.service.ts

```diff
@@ -17,383 +17,268 @@
-      if (this.isMultipleIncorrectSubmissionsIssue()) {
-        PlaythroughService.playthrough.issueType = (
-          AppConstants.ISSUE_TYPE_MULTIPLE_INCORRECT_SUBMISSIONS);
-        PlaythroughService.playthrough.issueCustomizationArgs = {
-          state_name: {
-            value: PlaythroughService.multipleIncorrectStateName.state_name
-          },
-          num_times_answered_incorrectly: {
-            value: (
-              PlaythroughService.multipleIncorrectStateName.num_times_incorrect)
-          }
-        };
-      } else if (this.isCyclicStateTransitionsIssue()) {
-        PlaythroughService.playthrough.issueType = (
-          AppConstants.ISSUE_TYPE_CYCLIC_STATE_TRANSITIONS);
-        PlaythroughService.playthrough.issueCustomizationArgs = {
-          state_names: {
-            value: PlaythroughService.cycleIdentifier.cycle.split(',')
-          }
-        };
-      } else if (this.isEarlyQuitIssue(timeSpentInExpInSecs)) {
-        PlaythroughService.playthrough.issueType = (
-          AppConstants.ISSUE_TYPE_EARLY_QUIT);
-        PlaythroughService.playthrough.issueCustomizationArgs = {
-          state_name: {
-            value:
-              PlaythroughService.playthrough.actions[
-                PlaythroughService.playthrough.actions.length - 1
-              ].actionCustomizationArgs.state_name.value
-          },
-          time_spent_in_exp_in_secs: {
-            value: timeSpentInExpInSecs
-          }
-        };
+    if (this.currentStateName() !== destStateName) {
+      const cycle = this.makeCycle(this.visitedStates.indexOf(destStateName));
+      if (angular.equals(this.cycle, cycle)) {
+        this.numCycles += 1;
+      } else {
+        this.cycle = cycle;
+        this.numCycles = 1;
       }
     }
+    this.visitedStates = [destStateName];
+  }
+}
 
-    private storePlaythrough(isNewPlaythrough: boolean): void {
-      let playthroughId = (
-        isNewPlaythrough ? null : PlaythroughService.playthrough.playthroughId);
-      let promise = this.http.post(this.getFullPlaythroughUrl(), {
-        playthrough_data: PlaythroughService.playthrough.toBackendDict(),
-        issue_schema_version: ServicesConstants.CURRENT_ISSUE_SCHEMA_VERSION,
-        playthrough_id: playthroughId
-      }).toPromise();
-      if (isNewPlaythrough) {
-        promise.then((
-            response: {
-                // eslint-disable-next-line camelcase
-                playthrough_stored: boolean, playthrough_id: string }) => {
-          if (response.playthrough_stored) {
-            // In cases where maximum number of playthroughs already exists, the
-            // above flag is not True and playthrough ID is not set.
-            PlaythroughService.playthrough.playthroughId = (
-              response.playthrough_id);
-          }
-        });
-      }
-    }
+class EarlyQuitTracker {
+  constructor(
+      public stateName: string = null,
+      public timeSpentInStateSecs: number = null) {}
```

Comment:
Just checking this is correct? (It was changed to timeSpentInStateMsecs in a different file above.)